firefox-ios/Client/Frontend/Browser/Tabs/Views/TabDisplayPanel.swift

```diff
@@ -66,6 +66,17 @@ class TabDisplayPanel: UIViewController,
         subscribeToRedux()
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        tabDisplayView.layoutIfNeeded()
+
+        let didLoadAction = TabPanelViewAction(panelType: panelType,
+                                               windowUUID: windowUUID,
+                                               actionType: TabPanelViewActionType.tabPanelDidLoad)
```

Comment:
Minor (and possibly annoying) nit: but are these really equivalent? Usually didLoad is treated or at least interpreted slightly differently from willAppear (the former happening once during UI creation, the latter potentially happening repeatedly during the lifecycle). If we really want the `tabPanelDidLoad` action to run on every appearance I wonder if it would be worth renaming the action to be more clear.

Reply:
Actually, I think this is a good point, `viewDidLoad` and `viewWillAppear` are very much not equivalent, so I don't think we should dispatch actions as if they are. I will create a new action.

This comment also got me thinking, so I played around with the fix a bit more and noticed that `viewWillAppear` gets called every time you drag the sheet, even slightly (see video), resulting in the action being dispatched more than just on first appearance


https://github.com/user-attachments/assets/b6680f47-34ad-40ef-a02b-d4d25b95599d